reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -64,51 +65,75 @@ def check_cloudwatch_log_group_tag(
+    log_group_list = check_cloudwatch_log_group_tag(log_groups, log_client)
+    return log_group_list
+
+
 def run(dry_run: bool, thread_pool_size: int, defer: Optional[Callable] = None) -> None:
-    aws_accounts = get_aws_accounts(cleanup=True)
-    for aws_acct in aws_accounts:
-        if aws_acct.get("cleanup"):
-            cloudwatch_cleanup_list = get_app_interface_cloudwatch_retention_period(
-                aws_acct
-            )
-            settings = queries.get_secret_reader_settings()
-            accounts = queries.get_aws_accounts(uid=aws_acct.get("uid"))
-            awsapi = AWSApi(1, accounts, settings=settings, init_users=False)
-            log_groups = awsapi.get_cloudwatch_logs(aws_acct)
-            session = awsapi.get_session(aws_acct["name"])
-            region = aws_acct["resourcesDefaultRegion"]
-            log_client = awsapi.get_session_client(session, "logs", region)
-            log_group_list = check_cloudwatch_log_group_tag(log_groups, log_client)
-
-            for cloudwatch_cleanup_entry in cloudwatch_cleanup_list:
-                for log_group in log_group_list:
-                    group_name = log_group["logGroupName"]
-                    retention_days = log_group.get("retentionInDays")
-                    regex_pattern = re.compile(cloudwatch_cleanup_entry.log_regex)
-                    if regex_pattern.match(group_name):
-                        log_group_tags = log_group["tags"]
-                        if not all(
-                            item in log_group_tags.items()
-                            for item in MANAGED_TAG.items()
-                        ):
-                            logging.info(
-                                f"Setting tag {MANAGED_TAG} for group {group_name}"
-                            )
-                            if not dry_run:
-                                awsapi.create_cloudwatch_tag(
-                                    aws_acct, group_name, MANAGED_TAG
-                                )
-                        if retention_days != int(
-                            cloudwatch_cleanup_entry.log_retention_day_length
-                        ):
-                            logging.info(
-                                f" Setting {group_name} retention days to {cloudwatch_cleanup_entry.log_retention_day_length}"
-                            )
-                            if not dry_run:
-                                awsapi.set_cloudwatch_log_retention(
-                                    aws_acct,
-                                    group_name,
-                                    int(
-                                        cloudwatch_cleanup_entry.log_retention_day_length
-                                    ),
-                                )
+    app_interface_aws_accounts = get_aws_accounts(cleanup=True)
+    with create_awsapi_client(app_interface_aws_accounts, 1) as awsapi:
```

Comment:
Now that we only create in `awsapi` in `run`, can pass `thread_pool_size` to it to speed up init

```python
with create_awsapi_client(app_interface_aws_accounts, thread_pool_size) as awsapi:
```